File: Lib/Telerik/Examples/QuickStart/GridView/Rows/RowDetails/RowDetailsViewDefinition.vb
﻿Imports System.Text
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.GridView.Rows.RowDetails
	Public Class RowDetailsViewDefinition
		Inherits TableViewDefinition
		Public Overrides Function CreateViewUIElement(ByVal viewInfo As GridViewInfo) As IRowView
			Dim tableElement As GridTableElement = CType(MyBase.CreateViewUIElement(viewInfo), GridTableElement)
			tableElement.ViewElement.RowLayout = CreateRowLayout()
			Return tableElement
		End Function

		Public Overrides Function CreateRowLayout() As IGridRowLayout
			Return New RowDetailsLayout()
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/GridView/Rows/RowDetails/RowDetailsViewDefinition.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.GridView.Rows.RowDetails
{
    public class RowDetailsViewDefinition : TableViewDefinition
    {
        public override IRowView CreateViewUIElement(GridViewInfo viewInfo)
        {
            GridTableElement tableElement = (GridTableElement)base.CreateViewUIElement(viewInfo);
            tableElement.ViewElement.RowLayout = CreateRowLayout();
            return tableElement;
        }

        public override IGridRowLayout CreateRowLayout()
        {
            return new RowDetailsLayout();
        }
    }
}
